audio-worklet/src/noisegate-script-processor.js

```diff
@@ -0,0 +1,217 @@
+   *                                 is zero.
+   * @param {Number} options.bufferSize the size of an onaudioprocess window.
+   *                                    The default lets the script processor
+   *                                    decide.
+   * @param {Number} options.timeConstant seconds for envelope follower's
+   *                                   smoothing filter delay. The
+   *                                   default has been set experimentally to
+   *                                   0.0025s.
+   * @param {Number} options.threshold decibel level beneath which sound is
+   *                                   muted. The default is 0 dBFS.
+   */
+  constructor(context, options) {
+    if (!(context instanceof BaseAudioContext))
+      throw 'Not a valid audio context.';
+    if (!options) options = {};
+    
+    const numberOfChannels = options.numberOfChannels || 1;
+    if (numberOfChannels > 2)
+      throw 'The maximum supported number of channels is two.';
+
+    this.context_ = context;
+    const bufferSize = options.bufferSize || 0;
+    this.threshold = options.threshold || 0;
+    this.attack = options.attack || 0;
+    this.release = options.release || 0;
+    
+    // The time constant of the filter has been set experimentally to balance
+    // roughly delay for high frequency suppression.
+    const timeConstant = options.timeConstant || 0.0025;
+
+    // Alpha controls a tradeoff between the smoothness of the
+    // envelope and its delay, with a higher value giving more smoothness at
+    // the expense of delay and vice versa.
+    this.alpha_ = NoiseGate.getAlphaFromTimeConstant_(
+        timeConstant, this.context_.sampleRate);
+
+    this.noiseGateKernel_ = this.context_.createScriptProcessor(
+        bufferSize, numberOfChannels, numberOfChannels);
+    this.noiseGateKernel_.onaudioprocess = this.onaudioprocess_.bind(this);
+
+    // The noise gate is connected to and from by dummy input and output nodes.
+    this.input = new GainNode(this.context_);
+    this.output = new GainNode(this.context_);
+    this.input.connect(this.noiseGateKernel_).connect(this.output);
+
+    // The previous envelope level, a float representing signal amplitude.
+    this.previousLevel_ = 0;
+
+    // The last weight (between 0 and 1) assigned, where 1 means the gate
+    // is open and 0 means it is closed and the sample in the output buffer is
+    // muted. When attacking, the weight will linearly decrease from 1 to 0, and
+    // when releasing the weight linearly increase from 0 to 1.
+    this.previousWeight_ = 1.0;
+    this.channel_ = new Float32Array(this.noiseGateKernel_.bufferSize);
+    this.envelope_ = new Float32Array(this.noiseGateKernel_.bufferSize);
+    this.weights_ = new Float32Array(this.noiseGateKernel_.bufferSize);
+  }
+
+  /**
+   * Gradually mutes input which registers beneath specified threshold.
+   * @param {AudioProcessingEvent} event object containing
```

Comment:
"An `Event` object containing input and output buffers."

Why does this comment span two lines?